src/main/java/th/ac/kmitl/science/comsci/example/nameprinters/NamePrinterLoader.java

```diff
@@ -0,0 +1,85 @@
+package th.ac.kmitl.science.comsci.example.nameprinters;
+
+import org.reflections.Reflections;
+
+import java.io.PrintWriter;
+import java.lang.reflect.Modifier;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Set;
+
+public class NamePrinterLoader implements Iterable<NamePrinter> {
+
+    private static NamePrinterLoader instance;
+    private static Reflections reflections;
+    private final String packageIdentifier;
+    private List<NamePrinter> namePrinters;
+    private Set<Class<? extends NamePrinter>> allNamePrinterClasses;
+
+    private NamePrinterLoader() {
+        packageIdentifier = getClass().getPackage().getName();
+        reflections = new Reflections(packageIdentifier);
+        setAllNamePrinterClasses(reflections.getSubTypesOf(NamePrinter.class));
+        namePrinters = getAvailableNamePrinters();
+    }
+
+    public static NamePrinterLoader getInstance() {
+        if (instance == null)
+            instance = new NamePrinterLoader();
+
+        return instance;
+    }
+
+    private void reload() {
+        namePrinters = getAvailableNamePrinters();
+    }
+
+    @Override
+    public Iterator<NamePrinter> iterator() {
+        return namePrinters.iterator();
+    }
+
+    private List<NamePrinter> getAvailableNamePrinters() {
+        List<NamePrinter> namePrinters = new ArrayList<>();
+        Set<Class<? extends NamePrinter>> allNamePrinterClasses = getAllNamePrinterClasses();
+
+        for (Class<? extends NamePrinter> namePrinterClass : allNamePrinterClasses) {
+            if (isInstantiable(namePrinterClass)) {
+                Object namePrinterObj;
+                try {
+                    namePrinterObj = namePrinterClass.newInstance();
+                    namePrinters.add((NamePrinter) namePrinterObj);
+                } catch (InstantiationException e) {
+                    continue;
```

Comment:
This `catch` statement shouldn't be skipped by `continue` you should throw message or stack trace from this catch.

Reply:
@khemtat I'll `printStackTrace()` then handle `continue` in `finally` statement